Fix 12+ failed RPC calls to calculate_assignee_distribution

## Problem

The workspace issues page makes 12+ failed RPC calls on every page load, each returning 404 errors. This adds significant network overhead and causes UI stuttering.

**User Impact:** 
- 100-200ms latency per failed request
- With triple initialization bug, this becomes 36+ failed requests
- Noticeable UI stuttering during data load

**Evidence from console logs:**
```
[ERROR] Failed to load resource: the server responded with a status of 404
@ https://egcxzonpmmcirmgqdrla.supabase.co/rest/v1/rpc/calculate_assignee_distribution  (×12+)

[WARNING] RPC function not found, falling back to client-side calculation  (×12+)
```

## Root Cause

The `calculate_assignee_distribution` RPC function doesn't exist in the Supabase schema. Code attempts RPC call, fails with 404, then falls back to client-side calculation.

**File:** `src/hooks/useAssigneeDistribution.ts:78`

## Proposed Solutions

### Option 1: Remove RPC Pattern (Recommended)

```typescript
export function useAssigneeDistribution({ excludeBots, limit, repositoryIds }) {
  return useQuery({
    queryKey: ['assignee-distribution', { excludeBots, limit, repositoryIds }],
    queryFn: async () => {
      // Skip RPC, go straight to client-side calculation
      const { data: assignees } = await supabase
        .from('issue_assignees')
        .select('*, issues(*)')
        .in('repository_id', repositoryIds);

      return calculateDistribution(assignees, excludeBots, limit);
    },
    staleTime: 5 * 60 * 1000, // Cache for 5 minutes
  });
}
```

### Option 2: Create Missing RPC Function

Create the RPC function in Supabase if server-side calculation is preferred for performance.

## Expected Impact

- 100% elimination of 404 errors (12+ → 0)
- Faster page load (removes 1-2 seconds of network overhead)
- Cleaner console logs
- Reduced server load

## Acceptance Criteria

- [ ] No 404 errors for `calculate_assignee_distribution` in Network tab
- [ ] No "RPC function not found" warnings in console
- [ ] Assignee distribution chart still displays correctly
- [ ] Data accuracy maintained

## Dependencies

- Should be fixed after #1186 (triple initialization) to see full impact

## Related

See full audit: `tasks/workspace-issues-performance-audit.md`

--- a/src/hooks/useAssigneeDistribution.ts
+++ b/src/hooks/useAssigneeDistribution.ts
@@ -24,8 +24,9 @@ interface UseAssigneeDistributionResult {
 }
 
 /**
- * Fallback client-side calculation when RPC function is not available
- * This happens when the migration hasn't been applied yet (e.g., deploy previews)
+ * Fallback client-side calculation for assignee distribution
+ * Used when RPC function is unavailable
+ * Fetches open issues and calculates distribution in the browser
  */
 async function fetchDistributionClientSide(
   repositoryIds: string[],
@@ -98,8 +99,9 @@ async function fetchDistributionClientSide(
 }
 
 /**
- * Hook to efficiently fetch assignee distribution using database-side aggregation
- * This optimizes performance by moving computation from client to database
+ * Hook to fetch assignee distribution for issues
+ * Uses database-first approach with RPC for optimal performance
+ * Falls back to client-side calculation if RPC is unavailable
  */
 export function useAssigneeDistribution({
   repositoryIds,
@@ -123,6 +125,7 @@ export function useAssigneeDistribution({
       setLoading(true);
       setError(null);
 
+      // Try database-first approach with RPC function
       const { data: result, error: rpcError } = await supabase.rpc(
         'calculate_assignee_distribution',
         {
@@ -132,17 +135,13 @@ export function useAssigneeDistribution({
         }
       );
 
-      // If RPC function doesn't exist (migration not applied), fallback to client-side calculation
+      // If RPC function doesn't exist, fallback to client-side calculation
       if (rpcError && rpcError.message.includes('Could not find the function')) {
         console.warn(
           'RPC function not found, falling back to client-side calculation:',
           rpcError.message
         );
-        const fallbackData = await fetchDistributionClientSide(
-          repositoryIds,
-          excludeBots,
-          limit
-        );
+        const fallbackData = await fetchDistributionClientSide(repositoryIds, excludeBots, limit);
         setData(fallbackData);
         return;
       }

